pandas/_libs/window.pyx

```diff
@@ -1391,7 +1392,19 @@ def roll_quantile(ndarray[float64_t, cast=True] input, int64_t win,
 
         if nobs >= minp:
             idx = int(quantile * <double>(nobs - 1))
-            output[i] = skiplist.get(idx)
+
+            # Exactly last point
+            if idx == nobs - 1:
```

Comment:
This can only happen if `quantile == 1.0`, right? It might be worth removing the branch from the inner loop: I think we could actually redirect `quantile == 1.0` to `roll_max` and `quantile == 0.0` to `roll_min`? If I am not missing anything, these are asymptotically faster, O(n) vs O(n log w).